src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -85,6 +95,22 @@ public void AllowStop()
 
         public void Stop(TimeSpan timeout)
         {
+            // Close and wait for all connections
+            if (!ConnectionManager.WalkConnectionsAndClose(timeout))
+            {
+                _log.LogError(0, null, "Waiting for connections timed out");
+            }
+
+            PostAsync(state =>
+            {
+                var listener = (KestrelThread)state;
+                var writeReqPool = listener.WriteReqPool;
```

Comment:
Why not just `listener.WriteReqPool.Dispose()`?

Also nit: remove blank line at the end of the block.